L2.py

```diff
@@ -0,0 +1,80 @@
+from numpy import *
+from matplotlib.pyplot import*
+from random import *
+import time
+def tab():
+    tab=[]
+    tab2=[]
+    size=randint(2,10)
+    for i in range(size):
+        tab.append(randint(-9,9))
+    return tab
+def bucket(tab):
+    clock_start = time.perf_counter()
+    t1=tab
+    s=len(t1)
+    mx=max(t1)+1
+    mi=min(t1)
+    j=0
+    t=0
+    t2=[]
+    ros=[]
+    mal=[]
+    for i in range (mi,mx):
+        t=0
+        j=1
+        while (j<=s):
+            if (t1[j-1]==i):
+                t2.append(i)
+                j=s
+            j=j+1
+            if (j>s):
+                break
+    for i in range(len(t2)):
+        for j in range(len(t1)):
+            if t2[i]==t1[j]:
+                ros.append(t1[j])
+    for k in range(len(ros)):
+        mal.append(ros[len(ros)-k-1])
+    clock_stop = time.perf_counter()
+    print("time",(clock_stop-clock_start)/0.10)
+    return mal
+
+def bubble(tab):
+    clock_start = time.perf_counter()
+    r1=tab.copy()
+    s = len(r1)
+    for i in range(s):
+        for j in range(0, s-i-1):
+            if r1[j] > r1[j+1] :
+                r1[j], r1[j+1] = r1[j+1], r1[j]
+    clock_stop = time.perf_counter()
```

Comment:
mierzenie czasu raczej nie powinno być zaszyte w samym algorytmie sortującym. Alg. piszemy po to by nam sortował. To że chcemy zbadać jego wydajność to na pewno nie chcemy tego robić za każdym razem gdy chcemy cos po prostu posortować. To powinno byc rozdzielone.